### PR TITLE
test(vectored-io): add comprehensive vectored I/O edge-case test suite

### PR DESCRIPTION
**Pull request by CharlieVinnie**
Adds `test-vectored-io` covering `readv`, `writev`, `pread64`, `pwrite64`, `preadv`, `pwritev`, `preadv2`, and `pwritev2` — 66 test cases across 12 sections: normal scatter/gather, EBADF, EINVAL (iovcnt & negative offset), ESPIPE on pipes, zero-length iovecs, directory fd interactions, partial reads, EISDIR, writev atomicity on pipes, and EPIPE on broken pipes.

Merges two prerequisite bugfix branches:
- `agent-bugfix-preadv-pwritev-negative-offset` — preadv2/pwritev2 offset=-1 support, unsupported flags rejection, and preadv/pwritev negative offset validation, [https://github.com/rcore-os/tgoskits/pull/326](https://github.com/rcore-os/tgoskits/pull/326)
- `agent-bugfix-writev-dir-ebadf` — writev/pwrite on directory fd returns EBADF, [https://github.com/rcore-os/tgoskits/pull/324](https://github.com/rcore-os/tgoskits/pull/324)

These two pull requests should be merged before this one.

### CODE REVIEW

**Review by ZR233 — APPROVED** [review]
这组 vectored I/O 修复和回归用例基本把关键边界都补到了，当前没看到阻塞合入的问题。

File: test-suit/starryos/normal/test-vectored-io/c/src/main.c [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+    /* 3.1 readv with iovcnt=0 => returns 0 (Linux behavior) */
+    n = readv(fd, iov, 0);
+    CHECK_RET(n, 0, "3.1 readv iovcnt=0 => 0");
+
+    /* 3.2 writev with iovcnt=0 => returns 0 */
+    n = writev(fd, iov, 0);
+    CHECK_RET(n, 0, "3.2 writev iovcnt=0 => 0");
+
+    /* 3.3 readv with iovcnt=-1 => EINVAL */
+    CHECK_ERR(readv(fd, iov, -1), EINVAL, "3.3 readv iovcnt=-1 => EINVAL");
```
Comment (ZR233) [inline test-suit/starryos/normal/test-vectored-io/c/src/main.c]: 负 `iovcnt` 用 libc `readv` 会被带 access attribute 的头文件诊断为 warning，并在 `-Werror` 下构建失败；建议用 `syscall(SYS_readv, ...)` 覆盖该错误路径。

File: test-suit/starryos/normal/test-vectored-io/c/src/main.c [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+
+    /* 3.4 writev with iovcnt=-1 => EINVAL */
+    iov[0].iov_base = dummy; iov[0].iov_len = 4;
+    CHECK_ERR(writev(fd, iov, -1), EINVAL, "3.4 writev iovcnt=-1 => EINVAL");
+
+    /* 3.5 readv with iovcnt > IOV_MAX => EINVAL */
+    CHECK_ERR(readv(fd, iov, IOV_MAX + 1), EINVAL, "3.5 readv iovcnt>IOV_MAX => EINVAL");
```
Comment (ZR233) [inline test-suit/starryos/normal/test-vectored-io/c/src/main.c]: `IOV_MAX + 1` 搭配只有 3 项的 `iov` 会被诊断为越界读取；建议使用 raw syscall，或准备足够大的 iovec 数组后再测内核的 `EINVAL`。

**Review by ZR233 — CHANGES_REQUESTED** [review]
重新拉取当前 head 复核后，不建议维持之前的 approve。该测试覆盖四架构配置，但 CMake 开启 `-Werror` 后，源码里对 libc `readv/writev` wrapper 的负 iovcnt / 超大 iovcnt 用例会被现代 GCC 诊断打断，存在构建不过的风险。

File: test-suit/starryos/normal/test-vectored-io/c/src/main.c (hunk 1 above)
Comment (Charliewzy) [reply test-suit/starryos/normal/test-vectored-io/c/src/main.c]: 谢谢，已使用 raw syscall 来应对该问题。

File: test-suit/starryos/normal/test-vectored-io/c/src/main.c (hunk 2 above)
Comment (Charliewzy) [reply test-suit/starryos/normal/test-vectored-io/c/src/main.c]: 谢谢，已使用 raw syscall 来应对该问题。

**Comment by CharlieVinnie** [comment]
> 重新拉取当前 head 复核后，不建议维持之前的 approve。该测试覆盖四架构配置，但 CMake 开启 `-Werror` 后，源码里对 libc `readv/writev` wrapper 的负 iovcnt / 超大 iovcnt 用例会被现代 GCC 诊断打断，存在构建不过的风险。

谢谢，已使用 raw syscall 来应对该问题。

**Comment by AsakuraMizu** [comment]
@octopusreview

**Review by ZR233 — APPROVED** [review]
